src/main/kotlin/gg/essential/elementa/ElementaVersion.kt

```diff
@@ -28,6 +28,21 @@ enum class ElementaVersion {
      */
     V1,
 
+    /**
+     * This Elementa version improves the behavior of mouse input in two ways
+     *
+     * 1. In Minecraft versions <=1.12.2, the game calculates the position of the mouse input in an integer context
+     * relative to scaled pixels. However, Elementa uses real pixels to determine the position of components leading
+     * to situations where certain components or parts of components are not clickable due the game truncating the decimal.
+     * This Elementa version improves this behavior by restoring the decimal component of mouse clicks to the mouse positions
+     * on affected versions.
+     *
+     * 2. Minecraft mouse click input is positioned in the top left corner of a pixel. As a result, the left and top pixel of
+     * a component do not register clicks and components with a width or height of 1 are also not clickable. This Elementa version
+     * improves this behavior by moving the click to the center of the pixel.
```

Comment:
It does this not only for clicks but also for `getMousePosition`. Should also mention that we're talking about real pixel here, not MC pixel. And as above, exactly what was changed, and what to look out for. And finally, it could also do with an example.
```suggestion
     * improves this behavior by offsetting mouse coordinates to the center of the real pixel.
     * In particular, [gg.essential.elementa.components.Window.mouseClick] will add half a real pixel to the passed
     * coordinates and [gg.essential.elementa.UIComponent.getMousePosition] will add half a real pixel to the returned
     * coordinates.
     * As a result, if you have previously relied on the exact distance between a click and a component, beware that
     * this value will now be off by up to half a real pixel compared to what it used to be. If required, you can get
     * back the original real-pixel-aligned value by rounding down to the nearest real pixel via
     * [gg.essential.elementa.utils.guiHint].
     * E.g. if the user clicks on the MC pixel at 3/4 with their GUI scale set to 2, the click used to be processed at
     * 3.0/4.0, but with this change it will appear at 3.25/4.25 (because 0.25 is half a real pixel at scale 2).
```